This fixes builds on uClibc and musl.  See
https://bugs.gentoo.org/show_bug.cgi?id=544756

This patch is a slight modification of uptream commits:
http://www.sudo.ws/repos/sudo/rev/86eb67f3c41a
http://www.sudo.ws/repos/sudo/rev/e0794f05e95c

diff -Naur sudo-1.8.12.orig/lib/util/getopt_long.c sudo-1.8.12/lib/util/getopt_long.c
--- sudo-1.8.12.orig/lib/util/getopt_long.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/lib/util/getopt_long.c	2015-04-10 19:21:20.337032782 +0000
@@ -52,6 +52,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #ifdef STDC_HEADERS
 # include <stdlib.h>
diff -Naur sudo-1.8.12.orig/lib/util/mksiglist.c sudo-1.8.12/lib/util/mksiglist.c
--- sudo-1.8.12.orig/lib/util/mksiglist.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/lib/util/mksiglist.c	2015-04-10 19:22:38.719856268 +0000
@@ -43,6 +43,7 @@
 #include "mksiglist.h"
 
     printf("#include <config.h>\n");
+    printf("#include <sys/types.h>\n");
     printf("#include <signal.h>\n");
     printf("#include \"sudo_compat.h\"\n\n");
     printf("const char *const sudo_sys_siglist[NSIG] = {\n");
diff -Naur sudo-1.8.12.orig/lib/util/mksigname.c sudo-1.8.12/lib/util/mksigname.c
--- sudo-1.8.12.orig/lib/util/mksigname.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/lib/util/mksigname.c	2015-04-10 19:22:10.738491394 +0000
@@ -43,6 +43,7 @@
 #include "mksigname.h"
 
     printf("#include <config.h>\n");
+    printf("#include <sys/types.h>\n");
     printf("#include <signal.h>\n");
     printf("#include \"sudo_compat.h\"\n\n");
     printf("const char *const sudo_sys_signame[NSIG] = {\n");
diff -Naur sudo-1.8.12.orig/lib/util/regress/fnmatch/fnm_test.c sudo-1.8.12/lib/util/regress/fnmatch/fnm_test.c
--- sudo-1.8.12.orig/lib/util/regress/fnmatch/fnm_test.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/lib/util/regress/fnmatch/fnm_test.c	2015-04-10 19:21:20.340032928 +0000
@@ -6,6 +6,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
 #ifdef HAVE_STRING_H
diff -Naur sudo-1.8.12.orig/lib/util/regress/glob/globtest.c sudo-1.8.12/lib/util/regress/glob/globtest.c
--- sudo-1.8.12.orig/lib/util/regress/glob/globtest.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/lib/util/regress/glob/globtest.c	2015-04-10 19:21:20.341032977 +0000
@@ -6,6 +6,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
 #ifdef HAVE_STRING_H
diff -Naur sudo-1.8.12.orig/lib/util/sha2.c sudo-1.8.12/lib/util/sha2.c
--- sudo-1.8.12.orig/lib/util/sha2.c	2015-02-09 18:40:09.000000000 +0000
+++ sudo-1.8.12/lib/util/sha2.c	2015-04-10 19:21:20.342033026 +0000
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013 Todd C. Miller <Todd.Miller@courtesan.com>
+ * Copyright (c) 2013-2015 Todd C. Miller <Todd.Miller@courtesan.com>
  *
  * Permission to use, copy, modify, and distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -25,6 +25,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #ifdef STDC_HEADERS
 # include <stdlib.h>
diff -Naur sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_base64.c sudo-1.8.12/plugins/sudoers/regress/parser/check_base64.c
--- sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_base64.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/plugins/sudoers/regress/parser/check_base64.c	2015-04-10 19:21:20.342033026 +0000
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013 Todd C. Miller <Todd.Miller@courtesan.com>
+ * Copyright (c) 2013-2015 Todd C. Miller <Todd.Miller@courtesan.com>
  *
  * Permission to use, copy, modify, and distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -16,6 +16,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #ifdef STDC_HEADERS
 # include <stdlib.h>
diff -Naur sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_digest.c sudo-1.8.12/plugins/sudoers/regress/parser/check_digest.c
--- sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_digest.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/plugins/sudoers/regress/parser/check_digest.c	2015-04-10 19:21:20.343033075 +0000
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013 Todd C. Miller <Todd.Miller@courtesan.com>
+ * Copyright (c) 2013-2015 Todd C. Miller <Todd.Miller@courtesan.com>
  *
  * Permission to use, copy, modify, and distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -16,6 +16,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #ifdef STDC_HEADERS
 # include <stdlib.h>
diff -Naur sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_hexchar.c sudo-1.8.12/plugins/sudoers/regress/parser/check_hexchar.c
--- sudo-1.8.12.orig/plugins/sudoers/regress/parser/check_hexchar.c	2015-02-09 18:40:10.000000000 +0000
+++ sudo-1.8.12/plugins/sudoers/regress/parser/check_hexchar.c	2015-04-10 19:21:20.344033124 +0000
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2014 Todd C. Miller <Todd.Miller@courtesan.com>
+ * Copyright (c) 2014-2015 Todd C. Miller <Todd.Miller@courtesan.com>
  *
  * Permission to use, copy, modify, and distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -16,6 +16,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <stdio.h>
 #ifdef STDC_HEADERS
 # include <stdlib.h>
diff -Naur sudo-1.8.12.orig/plugins/sudoers/solaris_audit.c sudo-1.8.12/plugins/sudoers/solaris_audit.c
--- sudo-1.8.12.orig/plugins/sudoers/solaris_audit.c	2015-02-09 18:53:54.000000000 +0000
+++ sudo-1.8.12/plugins/sudoers/solaris_audit.c	2015-04-10 19:21:20.345033172 +0000
@@ -15,6 +15,8 @@
  */
 
 #include <config.h>
+
+#include <sys/types.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
